src/app/pages/mainpage/mainpage-api.service.ts

```diff
@@ -0,0 +1,97 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { forkJoin, map, Observable } from 'rxjs';
+import { TrendingTimeWindow, TrendingType } from '../../types/trending-request';
+import { MediaListResponse } from 'src/app/types/media-list-response';
+import { GridMediaItem } from 'src/app/types/grid-media-item';
+import { MovieDTO } from 'src/app/types/DTO/movie-dto';
+import { TvShowDTO } from 'src/app/types/DTO/tv-show-dto';
+import { MediaItem } from 'src/app/types/media-item';
+import { DtoTransformService } from 'src/app/services/dto-transform.service';
+import { Movie } from 'src/app/types/movie';
+import { TvShow } from 'src/app/types/tv-show';
+import { MediaToGridMediaService } from 'src/app/services/media-to-grid-media.service';
+import { RESPONSE_PER_PAGE } from 'src/api-info';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class MainpageRequestsService {
+  constructor(
+    private readonly http: HttpClient,
+    private readonly dtoTranform: DtoTransformService,
+    private readonly mediaToGridMedia: MediaToGridMediaService
+  ) {}
+
+  requestTrending(
+    timeWindow: TrendingTimeWindow = TrendingTimeWindow.day
+  ): Observable<GridMediaItem[]> {
+    const trendingMoviesRequest$ = this.http.get<MediaListResponse<MovieDTO>>(
+      `trending/${TrendingType.movie}/${timeWindow}`
+    );
+    const trendingTvShowsRequest$ = this.http.get<MediaListResponse<TvShowDTO>>(
+      `trending/${TrendingType.tv}/${timeWindow}`
+    );
+
+    const joinedMoviesAndTvShows = this.joinMoviesAndTvShows(
+      trendingTvShowsRequest$,
+      trendingMoviesRequest$
+    );
+
+    return this.convertToGridMediaItem(joinedMoviesAndTvShows);
+  }
+
+  requestPopular(): Observable<GridMediaItem[]> {
+    const popularMoviesRequest$ =
+      this.http.get<MediaListResponse<MovieDTO>>('movie/popular');
+    const popularTvShowsRequest$ =
+      this.http.get<MediaListResponse<TvShowDTO>>('tv/popular');
+
+    const joinedMoviesAndTvShows = this.joinMoviesAndTvShows(
+      popularTvShowsRequest$,
+      popularMoviesRequest$
+    );
+
+    return this.convertToGridMediaItem(joinedMoviesAndTvShows);
```

Comment:
I would move it to UI layer. For example create some pipe, which map business entity to UI entity

Reply:
will think of solution

Reply:
Moved transforming to `GridMediaItem` to pipe, updated mainpage api and components.